subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationPublications.java

```diff
@@ -81,9 +86,14 @@ public DefaultConfigurationPublications(DisplayName displayName,
         this.fileCollectionFactory = fileCollectionFactory;
         this.attributesFactory = attributesFactory;
         this.domainObjectCollectionFactory = domainObjectCollectionFactory;
+        this.objectFactory = objectFactory;
         this.attributes = attributesFactory.mutable(parentAttributes);
     }
 
+    public void setObjectFactory(ObjectFactory objectFactory) {
```

Comment:
Once https://github.com/gradle/gradle/pull/18921 is in, please rebase/merge this PR and remove this